### ISSUE TITLE
Pull signed images from libraries returns non-zero exit status

### ISSUE THREAD

**Issue by vincentlabonte** [apptainer/singularity#3646]
## Version of Singularity:

Master branch (build: 5fac9d2).

### Expected behavior

Pulling signed images from libraries should return 0 for exit status if no errors occurred.

### Actual behavior

Pulling signed images from libraries returns 1 for exit status.

### Steps to reproduce behavior

`$ singularity pull library://sylabs/tests/signed:1.0.0`
INFO:    Key with ID EDECE4F3F38D871E not found in local keyring, downloading from keystore...
INFO:    Container is signed
Data integrity checked, authentic and signed by:
        Sylabs Admin <support@sylabs.io>, Fingerprint 8883491F4268F173C6E5DC49EDECE4F3F38D871E
WARNING: no local key matching entity
INFO:    Download complete: signed_1.0.0.sif
`$ echo $?`
1

**Comment by vincentlabonte** [comment]
The verify command also returns non-zero exit status:

`$ singularity verify ./signed_1.0.0.sif`
Verifying image: ./signed_1.0.0.sif
INFO:    Key with ID EDECE4F3F38D871E not found in local keyring, downloading from keystore...
INFO:    Container is signed
Data integrity checked, authentic and signed by:
        Sylabs Admin <support@sylabs.io>, Fingerprint 8883491F4268F173C6E5DC49EDECE4F3F38D871E
`$ echo $?`
1

**Comment by WestleyK** [comment]
The "exit status 1" is because you dont have that key locally (in your keyring file). You can do:

```
$ singularity key list
```
To list your public keys, and:

```
$ singularity key pull 8883491F4268F173C6E5DC49EDECE4F3F38D871E
```
To pull that key to your keyring.

**Comment by WestleyK** [comment]
This workflow/output should be improved in https://github.com/sylabs/singularity/pull/3615

**Comment by vincentlabonte** [comment]
Thanks @WestleyK! Feel free to close this issue

**Comment by WestleyK** [comment]
I'll leave it open for now, the workflow is still a bit confusing, will close when fixed :+1:

**Comment by GodloveD** [comment]
I agree with @vincentlabonte that this is confusing behavior.  

imho, non-zero exit status-es should be reserved for situations in which the code actually fails.  In the situations mentioned here, the code succeeds and does exactly what the user expects it to do.  An image is pulled or verified.  I think the non-zero exit status-es (stati?) are intended to signal to the user that unsigned images should be treated with care or that untrusted keys should be approached with caution.  But afaik, this is a very non-traditional way to use an exit status.

In short non-zero exit statuses should be reserved for actual errors.  If we want to warn users about potentially unwise decisions, let's find a less confusing way to do it.
